projects/addon-table/components/table-pagination/table-pagination.component.ts

```diff
@@ -108,15 +139,83 @@ export class TuiTablePaginationComponent {
         };
     }
 
+    /**
+     * @deprecated Use `selectPreviousPage()` instead.
+     */
     protected back(): void {
-        this.page--;
-        this.pageChange.emit(this.page);
-        this.paginationChange.emit(this.pagination);
+        this.selectPreviousPage();
     }
 
+    /**
+     * @deprecated Use `selectNextPage()` instead.
+     */
     protected forth(): void {
-        this.page++;
-        this.pageChange.emit(this.page);
+        this.selectNextPage();
+    }
+
+    protected selectFirstPage(): void {
+        this.selectPage(this.firstPage);
+    }
+
+    protected selectPreviousPage(): void {
+        const previousPage = this.previousPage;
+
+        if (previousPage === null) {
+            return;
```

Comment:
Early returns make no sense if it's just one line. This looks like it would be better to just support `null` in `selectPage` and do the early return there.